### ISSUE TITLE
Android: Crash when switching between two BottomSheets 

### ISSUE THREAD

**Issue by pijnappel** [the49ltd/The49.Maui.BottomSheet#37]
Since the last update the control crashes when I switch between two different BottomSheets.
I can open and close (and repeat) the first without a problem. Then I open and close the second; no problem. 
Then I try opening the first again. Exception. Stack trace below:

**System.ArgumentNullException:** 'Value cannot be null. (Parameter 'source')'

> at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
>    at System.Linq.OrderedEnumerable`2[[System.Collections.Generic.KeyValuePair`2[[The49.Maui.BottomSheet.Detent, The49.Maui.BottomSheet, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[System.Double, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Double, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..ctor(IEnumerable`1 source, Func`2 keySelector, IComparer`1 comparer, Boolean descending, OrderedEnumerable`1 parent)
>    at System.Linq.Enumerable.OrderByDescending[KeyValuePair`2,Double](IEnumerable`1 source, Func`2 keySelector)
>    at The49.Maui.BottomSheet.BottomSheetManager.LayoutDetents(BottomSheetBehavior behavior, ViewGroup container, IDictionary`2 heights, Double maxSheetHeight)
>    at The49.Maui.BottomSheet.BottomSheetController.Layout()
>    at The49.Maui.BottomSheet.BottomSheetController.<Show>b__30_0()
>    at Microsoft.Maui.Dispatching.Dispatcher.<>c__DisplayClass4_0.<DispatchImplementation>b__0() in D:\a\_work\1\s\src\Core\src\Dispatching\Dispatcher.Android.cs:line 22
>    at Java.Lang.Thread.RunnableImplementor.Run() in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:line 36
>    at Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net7.0/android-33/mcw/Java.Lang.IRunnable.cs:line 84
>    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 22

**Comment by paulvarache** [comment]
Please try rc5, and let me know if this fixes this

**Comment by pijnappel** [comment]
Tried it. The control does not crash anymore. On Android 13 everything works fine. On Android 9 the BottomSheet does sometimes hang or appear delayed when I try to reproduce the former error.

I think the source is a popup (from Community Toolkit) which my view opened when a certain error happened. 
I changed my implementation to get rid of the popup. Now everything works fine.

Thanks for the quick help!
